system/src/control/wifi_new.cpp

```diff
@@ -106,15 +108,29 @@ int joinNewNetwork(ctrl_request* req) {
     const NcpClientLock lock(ncpClient);
     CHECK(ncpClient->on());
     // FIXME: synchronize NCP client / NcpNetif and system network manager state
+    bool needToConnect = network_connecting(NETWORK_INTERFACE_WIFI_STA, 0, nullptr) ||
+            network_ready(NETWORK_INTERFACE_WIFI_STA, NETWORK_READY_TYPE_ANY, nullptr);
+    // To unblock
+    ncpClient->disable();
+    CHECK(ncpClient->enable());
+    // These two are in sync now
+    ncpClient->disconnect(); // ignore the error
     network_disconnect(NETWORK_INTERFACE_WIFI_STA, NETWORK_DISCONNECT_REASON_USER, nullptr);
-    CHECK(ncpClient->disconnect());
+    // FIXME: We are wiating for ncpNetif to potentially fully disconnect
+    HAL_Delay_Milliseconds(200);
```

Comment:
As we discussed, delay here should not be needed due to `NcpClientLock` and `processEvents()` call that requires a lock in esp32ncpnetif.